### PR TITLE
pipeline test report logging

### PR DESCRIPTION
**Pull request by zhenmichael**
console log failed tests from xunit reports generated by pipeline

Example of Build - client packages with failed tests
image

Example of Build - azure with no failing tests
image

Example of Build - benchmark-tool with no published tests in junit.xml format
image

### CODE REVIEW

File: tools/helper/printReport.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+const parseTestReport = (filename) => {
+    fs.readFile(filename,  'utf8', (err, data) => {
+        let failedTests;
+        parser.parseString(data, { mergeAttrs: true }, (err, res) => {
+            // console.log("READ FILE RESULTS: ", res)
+            failedTests = findFailedTests(res);
+        })
+
+        failedTests?.forEach((test) => {
+            console.log(test)
```
Comment (scarlettjlee) [inline tools/helper/printReport.js:32]: Color for the text would be nice here.

**Comment by scarlettjlee** [comment]
The other pipelines that should also get this are the real service stress and end-to-end (i.e. include-test-real-service.yml).

What does the console output look like with your change? Do you have an example?

**Comment by scarlettjlee** [comment]
@curtisman, does having this as a separate task meet the intended use case for the bug? [AB#1950](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/1950)

File: tools/pipelines/templates/build-npm-package.yml [hunk 2]
```diff
@@ -328,6 +328,51 @@ stages:
...
+              TargetFolder: '$(Build.ArtifactStagingDirectory)/testFiles'
+            condition: and(succeededOrFailed(), eq(variables['TestReportsExists'], 'true'))
+
+          # Install parser library
+          - task: Bash@3
+            displayName: Install xml2js
+            inputs:
+              targetType: inline
+              script: |
+                npm i xml2js
```
Comment (curtisman) [inline tools/pipelines/templates/build-npm-package.yml:356]: I am not enamored with just install it here without a lock file.  Could be unstable?

In general, I wonder if it is better if we use multi-mocha-reporter to emit both xml and json, so we don't need to parse XML?

File: tools/pipelines/templates/build-npm-package.yml [hunk 3]
```diff
@@ -328,6 +328,51 @@ stages:
...
+            condition: and(succeededOrFailed(), eq(variables['TestReportsExists'], 'true'))
+
+          # Console Log Failed Tests
+          - task: Bash@3
+            displayName: Log Failed Tests
+            inputs:
+              targetType: inline
+              script: |
+                failedTests=$(node printReports.js '$(Build.ArtifactStagingDirectory)/testFiles')
+                if [ $failedTests == [] ]
```
Comment (curtisman) [inline tools/pipelines/templates/build-npm-package.yml:367]: In your description, "Example of Build - client packages with failed tests", there seems to have a script error?

**Comment by zhenmichael** [comment]
Closing PR for updated draft PR with requested changes: 
https://github.com/microsoft/FluidFramework/pull/14029

**Pull request closed without merging**